tests/rptest/tests/write_caching_fi_e2e_test.py

```diff
@@ -0,0 +1,241 @@
+import collections
+import time
+
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.services.kgo_verifier_services import (
+    KgoVerifierConsumerGroupConsumer,
+    KgoVerifierProducer,
+    KgoVerifierSeqConsumer,
+)
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class WriteCachingFailureInjectionE2ETest(RedpandaTest):
```

Comment:
at some point, you also had a test that restarted only parts of the cluster, right?

A, B, C
restart B, restart C, partition(A) etc

that test isn't needed anymore?

Reply:
We still have that test but in another PR https://github.com/redpanda-data/redpanda/pull/17671

I have split the tests into lower level tests where we assert liveness and some semantics of the protocol (the other PR) and into e2e tests where we have long lived consumers (this one).

Adding an e2e test for that scenario doesn't have much value in my opinion and decided to have it only once.